rstudio/rsconnect
# Reorganize deployment functions

This PR doesn't actually change any code, but it moves things around into different files, in an effort to organize things a little more logically. I'm not sure if it succeeds at that, and am happy to close it if it doesn't.

Fixes https://github.com/rstudio/connect/issues/21517

Principles I tried to follow:
- User-facing functions that aren't also supporting functions for multiple functions go in their own files.
  - Functions that are called only by one function go in the same file
- Functions that are called by many functions go in `[area]Utils.R` files.

List of the changes:
- `appDependencies.R`: Renamed from `dependencies.R`, now only contains `appDependencies()`.
- `writeManifest.R` only contains `writeManifest()`.
- `deployApp.R` only contains `deployApp()`, `bundleApp()`, and other code that is only called by `deployApp()`.
- `bundle.R` contains functions that are shared between `deployApp()`, `writeManifest()`, and `appDependencies()`.

### QA Notes

These changes shouldn't affect any functionality. It's worthwhile trying to deploy a piece of content to Connect with this build to make sure that nothing is broken.

```r
> rsconnect::deployApp(appDir = "tests/testthat/shinyapp-simple", server = "localhost")
```

Then the created `rsconnect` directory from your copy of this repo.

```r
> unlink("tests/testthat/shinyapp-simple/rsconnect", recursive = TRUE)
```
